### ISSUE TITLE
[BUG] GCC-12 Building on Linux `anonymous struct with base classes`

### ISSUE THREAD

**Issue by TheFloatingBrain** [ezEngine/ezEngine#612]
**Bug description:**
Failed to compile due too: 

```C++
In file included from /root/workdir/ezEngine/Code/Engine/Foundation/Algorithm/HashHelperString.h:7,
                 from /root/workdir/ezEngine/Code/Engine/Foundation/Algorithm/Implementation/HashHelperString.cpp:3,
                 from /root/workdir/ezEngine/build/Code/Engine/Foundation/Algorithm/Implementation/unity_b7083668826d79f9bcf59e78007db470e141336f.cpp:8:
/root/workdir/ezEngine/Code/Engine/Foundation/Containers/HybridArray.h:43:10: error: anonymous struct with base classes
   43 |   struct : ezAligned<EZ_ALIGNMENT_OF(T)>
      |          ^
```

In multiple files

**How to Reproduce:**
1. Modify CMakeLists.txt in root directory to only build the ThirdParty and Engine modules (comment out the others)
2. `mkdir build && cd build`
3. `cmake .. -G "Unix Makefiles"
4. `make -j8`

**Dev-branch commit hash:**
04c185a0cdf4905a33d5aa00ce842a64da76e70c

**Notes**
I installed assimp using conan, (may have changed some cmake stuff, but I dont think so)

**Environment**
OS: Ubuntu 22.04 LTS x86_64 (`podman` container)
Compiler: g++ (Ubuntu 12-20220319-1ubuntu1) 12.0.1 20220319 (experimental) [master r12-7719-g8ca61ad148f]

**Comment by TheFloatingBrain** [comment]
Cant the [C++11 syntax](https://en.cppreference.com/w/cpp/language/alignof) be used here? Or are older C++ versions/compilers targeted?

**Comment by Ingrater** [comment]
I'm currently working on other linux related features, and as this is a purely cosmetical issue that would only get rid of the compiler warning and not have any other benefits, I don't know when I'll get to fixing this.
@TheFloatingBrain if you want you can do a PR where you replace the use of the ezAligend helper struct with the c++11 alignas.

**Comment by TheFloatingBrain** [comment]
> I'm currently working on other linux related features, and as this is a purely cosmetical issue that would only get rid of the compiler warning and not have any other benefits, I don't know when I'll get to fixing this. @TheFloatingBrain if you want you can do a PR where you replace the use of the ezAligend helper struct with the c++11 alignas.

It is an error on my machine, AFIK I dont have `-Wall` on. But sounds good, I will give it a shot! :) Do you have a linux fork? If so, is it building for you?

**Comment by Ingrater** [comment]
There is no separate linux fork. All linux code is checked into the main repository, and is already in the "develop" branch.
Yes, linux is building for me.
Running cmake under linux I use these additional cmake flags:
-DEZ_BUILD_EXPERIMENTAL_VULKAN=ON -DCMAKE_BUILD_TYPE=Debug -DEZ_COMPILE_ENGINE_AS_DLL=ON

Please note that both vulkan support and linux support are highly experimental and under active development at this stage and it is to be expected that you will run into various issues.

**Comment by Ingrater** [comment]
This was fixed in  #647
